nns/nns_contract.go

```diff
@@ -276,27 +282,37 @@ func Register(name string, owner interop.Hash160, email string, refresh, retry,
 			panic("TLD already exists")
 		}
 		storage.Put(ctx, tldKey, 0)
-	} else {
-		if tldBytes == nil {
-			panic("TLD not found")
-		}
-		if parentExpired(ctx, 1, fragments) {
-			panic("one of the parent domains is not registered")
-		}
-		parentKey := getTokenKey([]byte(name[len(fragments[0])+1:]))
+		putNameStateWithKey(ctx, getTokenKey([]byte(name)), NameState{
+			Name: name,
+			// NNS expiration is in milliseconds
+			Expiration: int64(runtime.GetTime() + expire*1000),
+		})
+		putSoaRecord(ctx, name, email, refresh, retry, expire, ttl)
+		return true
```

Comment:
No notification will be emitted on registering TLDs which is correct, because it's not a token anymore. However, it still may be desired thing to see from the user side. Don't we want to notify about TLDs registering (probably, with some special notification) as far?

Reply:
currently in NeoFS nobody listens to these notifications. I don't see any practical need right now, but maybe u or @roman-khimov does?

Reply:
It's a good style in general to have events for significant state changes. But this particular case is not handled in the C# contract as well, so if we don't know how to use it, we can omit it for now.